packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -123,15 +125,49 @@ export async function executePlugins(
 
     progressBar?.updateTitle(`Executing  ${chalk.bold(pluginCfg.title)}`);
 
-    const pluginReport = await executePlugin(pluginCfg);
-    progressBar?.incrementInSteps(plugins.length);
-
-    return outputs.concat(pluginReport);
-  }, Promise.resolve([] as PluginReport[]));
+    try {
+      const pluginReport = await executePlugin(pluginCfg);
+      progressBar?.incrementInSteps(plugins.length);
+      return outputs.concat(Promise.resolve(pluginReport));
+    } catch (e: unknown) {
+      progressBar?.incrementInSteps(plugins.length);
+      return outputs.concat(
+        Promise.reject(e instanceof Error ? e.message : String(e)),
+      );
+    }
+  }, Promise.resolve([] as Promise<ExecutePluginResult>[]));
 
   progressBar?.endProgress('Done running plugins');
 
-  return pluginsResult;
+  const errorsCallback = ({ reason }: PromiseRejectedResult) =>
+    console.error(reason);
+
+  return Promise.allSettled(pluginsResult).then(results => {
```

Comment:
Maybe change to `await` instead of `then`?

```suggestion
const results = await Promise.allSettled(pluginsResult);
```

Reply:
Good point, fixed